File: lab06/echon.py
#!/usr/bin/env python3

import sys, re

if (len(sys.argv) != 3):
   print("Usage: %s <number of lines> <string>" % sys.argv[0])
   sys.exit()

try:
   n = int(sys.argv[1])
except:
   n = -1

if (n < 0):
   print("%s: argument 1 must be a non-negative integer" % sys.argv[0])

for i in range(n):
   print(sys.argv[2])

File: lab05/echon.pl
#!/usr/bin/perl -w

if (@ARGV != 2) {
   print "Usage: $0 <number of lines> <string>\n";
   exit;
}

if (!($ARGV[0] =~ /^\d+$/)) {
   print "$0: argument 1 must be a non-negative integer\n";
   exit; 
}

$n=0;
while ($n < $ARGV[0]) {
   print "$ARGV[1]\n";
   $n++;
}
